From 74c3838f66653be3d276ad722a91b1385f95a1d8 Mon Sep 17 00:00:00 2001
From: Helmut Grohne <helmut@subdivi.de>
Date: Tue, 26 Jun 2018 22:31:22 +0200
Subject: don't assume availability of libOggFLAC for cross compilation

Bug-Debian: https://bugs.debian.org/885857
---
 autoconf/libOggFLAC.m4 | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/autoconf/libOggFLAC.m4 b/autoconf/libOggFLAC.m4
index 5188e85..0fdbc96 100644
--- a/autoconf/libOggFLAC.m4
+++ b/autoconf/libOggFLAC.m4
@@ -58,7 +58,7 @@ dnl
 dnl Now check if the installed libOggFLAC is sufficiently new.
 dnl
       rm -f conf.libOggFLACtest
-      AC_TRY_RUN([
+      AC_COMPILE_IFELSE([
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
@@ -70,7 +70,7 @@ int main ()
   return 0;
 }
 
-],, no_libOggFLAC=yes,[echo $ac_n "cross compiling; assumed OK... $ac_c"])
+],, no_libOggFLAC=yes)
        CFLAGS="$ac_save_CFLAGS"
        LIBS="$ac_save_LIBS"
   fi
